view/newPerspective/utils.js

```diff
@@ -337,40 +301,97 @@ function getValuesObject(request, getPerspectiveUrl, handleHierarchyEvent, handl
        // when hierarchy is resolved in getHierarchy
       gotLens = true;
 
-      return res;
+      valuesObj.lens = res.body;
     });
 
-    const filterString  = getFilterQuery(valuesObj.perspective);
-    const getHierarchy = request('/v1/subjects/' +
-      valuesObj.perspective.rootSubject + '/hierarchy' + filterString)
+    const filterString  = getFilterQuery(perspective);
+    const getHierarchy = getPromiseWithUrl('/v1/subjects/' +
+    perspective.rootSubject + '/hierarchy' + filterString)
     .then((res) => {
 
       // if gotLens is false, hierarchyLoadEvent will be assigned
       // and NOT dispatched. Otherwise dispatch the hierarchy event
       hierarchyLoadEvent = handleHierarchyEvent(res.body, gotLens);
 
-      return res;
+      valuesObj.rootSubject = res.body;
     });
 
+    return [getLens, getHierarchy];
+  }
+
+  // Need to get all perspectives, for editing perspectives.
+  // for GET perspective, do NOT throw error if 404, since
+  // throwing an error skips loading the perspective picker.
+  // If successful, load the hierarchy and lens
+  const arr = [
+    getPromiseWithUrl('/v1/perspectives'),
+    getPromiseWithUrl(url)
+    .then((res) => {
+      Promise.all(getPageLoadingPromises(res.body));
+      return res;
+    })
+    .catch(console.log)
+  ];
+
+  return Promise.all(arr)
+  .then((responses) => {
+    valuesObj.perspectives = responses[0].body;
+
+    // use ternary as if pespective does not exist, responses[1] is undefined
+    const returnedPerspective = responses[1] ? responses[1].body : null;
+
+    // assign perspective-related values to the accumulator object
+    valuesObj.persNames = valuesObj.perspectives
+      .map((perspective) => perspective.name).sort();
+
+    /*
+     * One out of four situations can happen:
+     * GET named perspective: exists. Assign perspective
+     * GET default perspective: exists. Assign perspective
+     * GET named perspective: does NOT exist: handleError
+     * GET default perspective: does NOT exist: perspective is the first
+     *  perspective in the perspectives array. If no perspectives exist,
+     *  valuesObj.perspective = null
+     */
+     if (returnedPerspective) {
+        valuesObj.perspective = returnedPerspective;
+        valuesObj.name = valuesObj.perspective.name;
+     } else if (named) {
+
+        // named perspective does not exist
+        const name = url.split('/').pop();
+        customHandleError('Sorry. Perspective ' + name + ' not found.');
```

Comment:
"Sorry, but the perspective you were trying to load, _______, does not exist. Please try selecting a perspective from the dropdown."